server/tournaments/index.js

```diff
@@ -1073,7 +1074,7 @@ class Tournament extends Rooms.RoomGame {
 		this.completedMatches.add(room.id);
 		room.parent = null;
 		if (!room.battle) throw new Error("onBattleWin called without a battle");
-
+		if (!room.p1 || !room.p2) throw new Error("onBattleWin called with missing players");
```

Comment:
I think it makes sense to use `!` assertions, rather than add throws that don't add any information than the crash that would otherwise happen here.

I would even use `!` in place of the above `room.battle` check.

Reply:
Ohhh. Yeah, this is fine, then.

Reply:
(I might prefer `@ts-ignore`)

Reply:
I mean, `@ts-ignore` over 6 lines of code, or just an if-condition crash. I suppose you could reduce the amount of `@ts-ignore` by declaring a variable with `room.battle` as the value and disabling the compiler there, but that looks a bit awkward since there's already a `p1` variable.

Reply:
I thought it was only one line.

Reply:
I'd prefer an if condition over six lines of `ts-ignore`

Reply:
Wait, I was wrong, it would need 3 ``@ts-ignore``, which is still a line more than an if condition.